### PR TITLE
Add notification settings, which allows muting decky/plugin toast notifications

### PR DESCRIPTION
**Pull request by JohnnyCrazy**
Please tick as appropriate:
- [x] I have tested this code on a steam deck or on a PC
- [x] My changes generate no new errors/warnings
- [ ] This is a bugfix/hotfix
- [x] This is a new feature

If you're wanting to update a translation or add a new one, please use the weblate page: https://weblate.werwolv.net/projects/decky/

# Description

This fixes issue: #473 

image

### CODE REVIEW

File: frontend/src/components/DeckyState.tsx [hunk 1]
```diff
@@ -13,6 +14,7 @@ interface PublicDeckyState {
   hasLoaderUpdate?: boolean;
   isLoaderUpdating: boolean;
   versionInfo: VerInfo | null;
+  notifiationSettings: NotificationSettings;
```
Comment (jurassicplayer) [inline frontend/src/components/DeckyState.tsx:17]: `notifiationSettings` has a typo

File: frontend/src/components/DeckyState.tsx [hunk 2]
```diff
@@ -37,6 +40,7 @@ export class DeckyState {
       hasLoaderUpdate: this._hasLoaderUpdate,
       isLoaderUpdating: this._isLoaderUpdating,
       versionInfo: this._versionInfo,
+      notifiationSettings: this._notificationSettings,
```
Comment (jurassicplayer) [inline frontend/src/components/DeckyState.tsx:43]: same typo as above

Comment (JohnnyCrazy) [reply frontend/src/components/DeckyState.tsx]: Holy shit, there were a lot of typos lol. latest commit should fix it.

**Comment by ttang4299** [comment]
Hey @JohnnyCrazy , I don't know if there's room to implement a further change, but if you know how to toggle notifications for successful and/or unsuccessful update checks (or at least add a section where it can show when a successful update check was last performed), but if you know if this could be implemented, it would be very cool if you could add this whenever you have the time to

File: frontend/src/notification-service.tsx
Comment (AAGaming00) [inline frontend/src/notification-service.tsx:1]: I feel like this service stuff should be converted into something more generic for catching settings JS side, it's useful but a bit specific imo.

**Review by AAGaming00 — APPROVED** [review]

**Comment by Kalidibus** [comment]
@JohnnyCrazy  - Thank you so much for this! The notifications were a huge immersion breaker when in game.

**Pull request merged**